### PR TITLE
WIP: Introduce UserGroupRole model

### PR DESCRIPTION
**Pull request by dnlkoch**

### CODE REVIEW

File: src/main/java/de/terrestris/momo/security/MomoAuthenticationProvider.java [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+	/**
+	 * @param user
+	 * @param userGroups
+	 * @return
+	 */
+	private Set<Role> getAllUserRoles(User user) {
+		Set<Role> allUserRoles = new HashSet<Role>();
+
+		if (user != null) {
+			List<UserGroupRole> result = userGroupRoleDao.findUserRoles(user);
```
Comment (buehner) [inline src/main/java/de/terrestris/momo/security/MomoAuthenticationProvider.java:163]: I think you could also use the `findAllWhereFieldEquals` from SHOGun, i.e. `findAllWhereFieldEquals('user', user)`:

https://github.com/terrestris/shogun2/blob/master/src/shogun2-core/src/main/java/de/terrestris/shogun2/dao/GenericHibernateDao.java#L110

File: src/main/java/de/terrestris/momo/security/MomoAuthenticationProvider.java [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+	 */
+	@Autowired
+	private UserGroupRoleDao<UserGroupRole> userGroupRoleDao;
+
+	/**
+	 *
+	 */
+	@Autowired
+	private PasswordEncoder passwordEncoder;
+
```
Comment (buehner) [inline src/main/java/de/terrestris/momo/security/MomoAuthenticationProvider.java:59]: As the parent class `Shogun2AuthenticationProvider` has this field, too, why don't you remove this field here and call `getPasswordEncoder()` where you need it?

File: src/main/java/de/terrestris/momo/security/MomoAuthenticationProvider.java [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+			throw new UsernameNotFoundException(exceptionMessage);
+		} else if (!user.isActive()) {
+			LOG.warn("The user with the account name '" + accountName + "' is not active.");
+			throw new DisabledException(exceptionMessage);
+		} else {
+
+			encryptedPassword = user.getPassword();
+
+			// check if rawPassword matches the hash from db
+			if (passwordEncoder.matches(rawPassword, encryptedPassword)) {
```
Comment (buehner) [inline src/main/java/de/terrestris/momo/security/MomoAuthenticationProvider.java:100]: can't you use `getPasswordEncoder()` from parent class here?

**Comment by dnlkoch** [comment]
Closing this one, further implementation will be continued in [here](https://github.com/terrestris/momo3-backend/tree/user-group-entity-permissions).

**Pull request closed without merging**